File: MATLAB/src/utils/resolve_truth.m
function truth_path = resolve_truth(preferred_path)
%RESOLVE_TRUTH Locate the canonical truth file.
%   TRUTH_PATH = RESOLVE_TRUTH(PREFERRED_PATH) ensures that the truth file
%   used by MATLAB tasks is located at the canonical path
%   ``/Users/vimalchawda/Desktop/IMU/STATE_X001.txt``. The function returns
%   the path to the truth file or an empty string if the file could not be
%   found.

    if nargin==0 || isempty(preferred_path)
        here = fileparts(mfilename('fullpath'));
        repo = fileparts(fileparts(here)); % .../MATLAB/src/utils -> repo
        preferred_path = fullfile(repo,'DATA','Truth','STATE_X001.txt');
    end

    if isfile(preferred_path)
        fprintf('Using TRUTH: %s\n', preferred_path);
        truth_path = preferred_path;
        return;
    end

    warning('Truth file not found at preferred path.');
    truth_path = '';
end

File: src/utils/resolve_truth.py
"""Python analogue of ``MATLAB/src/utils/resolve_truth.m``."""

from __future__ import annotations

from pathlib import Path


def resolve_truth(preferred_path: str | None = None) -> str:
    """Return path to canonical truth file."""

    if not preferred_path:
        preferred_path = "/Users/vimalchawda/Desktop/IMU/STATE_X001.txt"
    preferred = Path(preferred_path)

    if preferred.is_file():
        print(f"Using TRUTH: {preferred}")
        return str(preferred)

    print("Truth file not found at preferred path.")
    return ""


__all__ = ["resolve_truth"]

